src/main/java/green/citibike/CitiFrame.java

```diff
@@ -0,0 +1,155 @@
+        mapViewer.addMouseMotionListener(mia);
+        mapViewer.addMouseListener(new CenterMapListener(mapViewer));
+        mapViewer.addMouseWheelListener(new ZoomMouseWheelListenerCursor(mapViewer));
+        mapViewer.addKeyListener(new PanKeyListener(mapViewer));
+
+        WaypointPainter<Waypoint> waypointPainter = new WaypointPainter<Waypoint>();
+
+        List<GeoPosition> track = new ArrayList<>();
+
+        RoutePainter routePainter = new RoutePainter(track);
+
+        List<org.jxmapviewer.painter.Painter<org.jxmapviewer.JXMapViewer>> painters = List.of(
+                routePainter,
+                waypointPainter
+        );
+
+        CompoundPainter<JXMapViewer> painter = new CompoundPainter<JXMapViewer>(painters);
+        mapViewer.setOverlayPainter(painter);
+
+        mapViewer.addMouseListener(new MouseAdapter() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                int x = e.getX();
+                int y = e.getY();
+                Point2D.Double point = new Point2D.Double(x, y);
+                GeoPosition position = mapViewer.convertPointToGeoPosition(point);
+                addToTrack(track, position, routePainter);
+                setWaypointPainter(track, waypointPainter);
+                mapViewer.repaint();
+            }
+        });
+
+        JButton next = new JButton("Map");
+        next.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                if (track.size() == 2) {
+                    Response response = mapPoints(track);
+                    textArea.setText(response.toString());
+                    setWaypointPainter(track, waypointPainter);
+                    mapViewer.zoomToBestFit(
+                            new HashSet<>(track),
+                            1.0
+                    );
+                }
+
+            }
+        });
+        main.add(next, BorderLayout.SOUTH);
+    }
+
+
+    private void addToTrack(List<GeoPosition> track, GeoPosition geoPos, RoutePainter rp) {
+        if (track.size() > 1) {
+            GeoPosition geoPosFirst = track.get(1);
+            track.clear();
+            track.add(geoPosFirst);
+        }
+
+        track.add(geoPos);
+        rp.setTrack(track);
+    }
+
+    private void setWaypointPainter(List<GeoPosition> track, WaypointPainter wp) {
+        HashSet<Waypoint> waypoints = new HashSet<>();
+
+        for (GeoPosition geoPos : track) {
+            waypoints.add(new DefaultWaypoint(geoPos));
+        }
+
+        wp.setWaypoints(waypoints);
+    }
+
+    private Response mapPoints(List<GeoPosition> track) {
+        Request request = new Request(track.get(0), track.get(1));
+        String json = request.toString();
+        APIGatewayProxyRequestEvent event = new APIGatewayProxyRequestEvent();
+        event.setBody(json);
+
+        Response response = requestHandler.handleRequest(event, null);
```

Comment:
When I say to use your lambda, what I mean is to make a request to your lambda running on AWS. 

Not what you are doing here.

Reply:
Okay, not sure I know how to do this -- what documentation can I look at for help (is it the s3 part)?

Reply:
You do know how to do this.
You did it last semester with Retrofit.

Reply:
Do you mean with the disposables? How does that interact with the lambda?

Reply:
You're going to make a network request to the lambda that you built that is running in the cloud.